java/server/src/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueue.java

```diff
@@ -100,6 +114,45 @@ public int getQueueSize() {
     }
   }
 
+  public String getQueueInfo() {
+    Lock readLock = lock.readLock();
+    readLock.lock();
+    try {
+      Map<String, BrowserInfo> browserCountMap = new HashMap<>();
+
+      List<Capabilities> capabilitiesList = sessionRequests.stream()
+        .map(SessionRequest::getHttpRequest)
+        .map(req -> {
+          try (
+            Reader reader = reader(req);
+            NewSessionPayload payload = NewSessionPayload.create(reader)) {
+            return payload.stream().iterator();
```

Comment:
Does this actually work? I'd expect `payload` and `reader` to be closed once we leave the scope of the `try` block.